src/corelibs/U2Core/src/util/U2SequenceUtils.cpp

```diff
@@ -463,23 +463,19 @@ void U2SequenceImporter::startSequence(U2OpStatus& os,
 
 void U2SequenceImporter::addBlock(const char* data, qint64 len, U2OpStatus& os) {
     CHECK(len > 0, );
-    // derive common alphabet
-    const DNAAlphabet* blockAl = U2AlphabetUtils::findBestAlphabet(data, len);
-    CHECK_EXT(blockAl != nullptr, os.setError("Failed to match sequence alphabet!"), );
-
-    const DNAAlphabet* oldAl = U2AlphabetUtils::getById(sequence.alphabet);
-    const DNAAlphabet* resAl = blockAl;
-    if (oldAl != nullptr) {
-        if (oldAl->getType() == DNAAlphabet_AMINO && resAl->getType() == DNAAlphabet_NUCL) {
-            resAl = oldAl;
-        } else if (resAl->getType() == DNAAlphabet_AMINO && oldAl->getType() == DNAAlphabet_NUCL) {
-            oldAl = resAl;
-        } else {
-            resAl = U2AlphabetUtils::deriveCommonAlphabet(blockAl, oldAl);
+    for (int i = 0; i < len; i++) {
+        if (Q_UNLIKELY(!alphabetCharacterHit[data[i]])) {
+            alphabetCharacterHit[data[i]] = true;
         }
-        CHECK_EXT(resAl != nullptr, os.setError(U2SequenceUtils::tr("Failed to derive sequence alphabet!")), );
     }
-
+    QByteArray bytes;
+    for (int i = 0; i < alphabetCharacterHit.size(); i++) {
+        if (Q_UNLIKELY(alphabetCharacterHit[i])) {
```

Comment:
Remove Q_UNLIKELY. This part is not a hotspot

Reply:
Why not? In most cases there are less hits than skips.

Reply:
If you can explain why we do not use Q_UNLIKELY with all other `if`s in our code you can explain this case too :)